### DISCUSSION TITLE
Better way to transform type in entity

### DISCUSSION THREAD

**Discussion by stefaanMLB** [mikro-orm/mikro-orm#5196] [Q&A]
Because we have historical DB schema, I often have to convert types in the entities, below an example from `tinyint` to `boolean`
```ts
  @Property({ type: 'number', length: 1, nullable: false, fieldName: 'visible' })
  _visible: number = 1

  @Property({ persist: false })
  get visible() {
    return this._visible === 1
  }
  set visible(value: boolean) {
    this._visible = value ? 1 : 0
  }
```

I wonder if it is possible to do the same in on go ?  So without having to define the `_visible` property.

**Comment by B4nan** [comment]
just define a custom type?

**Comment by stefaanMLB** [comment]
Yes ! works like a charm.
Many thanks !
